src/ui/components/common/content_actions/hide_post_button.rs

```diff
@@ -0,0 +1,34 @@
+use crate::{
+  serverfns::posts::HidePostAction,
+  ui::components::common::icon::{Icon, IconType},
+  utils::{traits::ToStr, types::Hidden},
+};
+use leptos::*;
+use leptos_router::ActionForm;
+
+#[component]
+pub fn HidePostButton(id: i32) -> impl IntoView {
+  let hide_post_action = expect_context::<HidePostAction>();
+  let hidden = expect_context::<Signal<Hidden>>();
+  let icon = Signal::derive(move || {
+    if hidden.get().0 {
+      IconType::Eye
+    } else {
+      IconType::EyeSlash
+    }
+  });
+
+  view! {
+    <li>
+      <ActionForm action=hide_post_action>
+        <input type="hidden" name="id" value=id />
+        <input type="hidden" name="hide" value=move || (!hidden.get().0).to_str() />
+        <button class="text-xs whitespace-nowrap" type="submit">
+          <Icon icon=icon class="inline-block" />
+          " "
+          {move || if hidden.get().0 { "Unhide post" } else { "Hide post" }}
```

Comment:
Not sure if you started i18n yet or not.

Reply:
I recall having trouble switching translations dynamically from user input. If we only want to use the language from the browser locale and not expose an option for users to change their interface language, we could probably go with i18n right away. Stilll have to check how it interacts with browser locale TBH.